sdk/src/azure/iot/az_iot_provisioning_client.c

```diff
@@ -558,3 +561,42 @@ AZ_NODISCARD az_result az_iot_provisioning_client_parse_received_topic_and_paylo
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_provisioning_client_get_request_payload(
+    az_iot_provisioning_client const* client,
+    az_span custom_payload_property,
+    az_iot_provisioning_client_payload_options const* reserved,
+    uint8_t* mqtt_payload,
+    size_t mqtt_payload_size,
+    size_t* out_mqtt_payload_length)
```

Comment:
should this actually be optional? I don't recall the json writer guaranteeing `NULL` termination and if someone doesn't provide this and the whole buffer is filled, they won't know that it isn't `NULL` terminated anymore (assuming zero'd out buffer).

Reply:
@ahsonkhan for comment here/tell me if I'm crazy

Reply:
>  I don't recall the json writer guaranteeing `NULL` termination

It doesn't.

You are right, @danewalton .

Since we are using the json_writer internally as an impl detail here, when this param is optional, there wouldn't be any way for the user to know how much was actually written to the buffer.

Reply:
Fixing.